net/pfSense-pkg-Quagga_OSPF/files/usr/local/pkg/quagga_ospfd.inc

```diff
@@ -513,22 +558,83 @@ function quagga_ospfd_plugin_carp($pluginparams) {
+        case "quaggadisable":
+            syslog(LOG_INFO, "Quagga: CARP mode {$ospfd_conf['carpmode']}");
+            $vip = get_configured_vip($ospfd_conf['carpstatusvid']);
+            syslog(LOG_INFO, "Quagga: CARP VHID and Interface {$vip['vhid']},{$vip['interface']}");
+            if ($vip['vhid'] != $vhid || $vip['interface'] != $friendly) {
+                return null;
+            }
+			
+            /* Start or stop the service as needed based on the CARP transition. */
+            if ((isset($ospfd_conf['carpmode']) && $ospfd_conf['carpmode'] == "quaggadisable") && (isset($ospfd_conf['carpstatusvid']) && $ospfd_conf['carpstatusvid'] != "none")) {
+                if ($pluginparams['event'] == "rc.carpmaster") {
+                    start_service("Quagga OSPFd");
+                } elseif ($pluginparams['event'] == "rc.carpbackup") {
+                    stop_service("Quagga OSPFd");
+                }
+            }
+            break;
+        case "ospfcost":
+            syslog(LOG_INFO, "Quagga: CARP mode {$ospfd_conf['carpmode']}");
+            $carpvips = explode(",",$ospfd_conf['carpcostvid']);
+            $match = FALSE;
+            $i = 0;
+            foreach ($carpvips as $vip) {
+                $vips[$i] = get_configured_vip($vip);
+                if (($vips[$i]['vhid'] == $vhid) && ($vips[$i]['interface'] == $friendly)){
+                    $match = TRUE;
+                    syslog(LOG_INFO, "Quagga: CARP VHID and Interface {$vips[$i]['vhid']} {$vips[$i]['interface']}");
+                    $phyint = get_real_interface($vips[$i]['interface']);
+                    syslog(LOG_INFO, "Quagga: CARP Interface identified {$phyint}");
+                    }
+                $i++;
+            }
+            unset($vip);
+
+            if ($match != TRUE){
+                syslog(LOG_NOTICE, "Quagga: Couldn't find an Interface or VHID match!");
+                return null;
+            }
+
+            /* Shift OSPF cost up or downas needed based on the CARP transition. */
+            $control_script = "/usr/local/bin/quaggactl";
+            if ((isset($ospfd_conf['carpmode']) && $ospfd_conf['carpmode'] == "ospfcost") && (isset($ospfd_conf['carpcostvid']) && substr($ospfd_conf['carpcostvid'],0,4) != "none")) {
+                if ($pluginparams['event'] == "rc.carpmaster") {
+                    syslog(LOG_INFO, "Quagga: Shifted OSPF Master cost on {$phyint} to {$ospfd_conf['carpactivecost']}");
+                    mwexec("{$control_script} ospf cost {$phyint} {$ospfd_conf['carpactivecost']}");
+                    return null;
+                } elseif ($pluginparams['event'] == "rc.carpbackup") {
+                    syslog(LOG_INFO, "Quagga: Shifted OSPF Backup cost on {$phyint} to {$ospfd_conf['carpbackupcost']}");
+                    mwexec("{$control_script} ospf cost {$phyint} {$ospfd_conf['carpbackupcost']}");
+                    return null;
+                }
+            }
+            break;
+    }
```

Comment:
This whole section still has leading spaces which should be tabs

Reply:
all fixed